commit e9d4b51968e0
Author: Samael Wang <freesamael@gmail.com>
Date:   Thu Sep 28 11:16:53 2017 +0800

    Bug 1401522 - Don't apply statusCode sent from parent if mCanceled or mStatus has been modified in the child. r=mayhemer, a=ritu
    
    When a WyciwygChannel is canceled, but WyciwygChannelParent::RecvCancel happens
    after WyciwygChannelParent::SendOnStartRequest, it would send statusCode=NS_OK
    to WyciwygChannelChild::OnStartRequest. So we should not apply the statusCode
    if mCanceled, just like how HttpChannelChild handles it.
    
    MozReview-Commit-ID: 5H3PUrlArIA
    
    --HG--
    extra : source : 8ebfa90c4ebca44ac1c3829ef54de74d3b004dfc
---
 netwerk/protocol/wyciwyg/WyciwygChannelChild.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git netwerk/protocol/wyciwyg/WyciwygChannelChild.cpp netwerk/protocol/wyciwyg/WyciwygChannelChild.cpp
index d14b6f925e37..85d3ebf24439 100644
--- netwerk/protocol/wyciwyg/WyciwygChannelChild.cpp
+++ netwerk/protocol/wyciwyg/WyciwygChannelChild.cpp
@@ -191,7 +191,9 @@ WyciwygChannelChild::OnStartRequest(const nsresult& statusCode,
 
   mState = WCC_ONSTART;
 
-  mStatus = statusCode;
+  if (!mCanceled && NS_SUCCEEDED(mStatus)) {
+    mStatus = statusCode;
+  }
   mContentLength = contentLength;
   mCharsetSource = source;
   mCharset = charset;
